doc/guides/gcov.rst

```diff
@@ -0,0 +1,44 @@
+.. _gcov:
+
+Coverage reports via GCOV
```

Comment:
Note that coverage can also be generated for the POSIX_ARCH based boards (which are used for testing without a real/emulated embedded target)
But this new documentation only talks about generating coverage in embedded targets as introduced in #9717.
I would think it would be logical to at least make this clear (that this only applies to embedded targets)